src/NHibernate/Tuple/DynamicMapInstantiator.cs

```diff
@@ -50,7 +50,7 @@ public object Instantiate()
 
 		protected virtual IDictionary GenerateMap()
 		{
-			return new Hashtable();
+			return new Dictionary<string, object>();
```

Comment:
I think it's better to create a new type `DynamicEntityInstantiator`.

Reply:
Why not.